code/modules/library/library_catalog.dm

```diff
@@ -0,0 +1,605 @@
+			"reports" = query.item[7],
+		))
+		flagged_books += CB
+	qdel(query)
+	return flagged_books
+
+/*
+  * # get_total_books
+  *
+  * External proc that counts the number of books in the DB that match the provided search parameters
+  * calling this with no arguments will return the complete count of books in the DB, if the query fails
+  * this proc will return null, so usages of this proc will need to account for that
+  *
+  * Arguments:
+  * * datum/library_user_data/search_terms - datum with parameters for what we want to query our DB for
+ */
+/datum/library_catalog/proc/get_total_books(datum/library_user_data/search_terms)
+	var/list/search_query = buildSearchQuery(search_terms)
+	var/sql = "SELECT COUNT(id) FROM library" + search_query[1]
+	var/list/sql_params = search_query[2]
+
+	var/datum/db_query/count_query = SSdbcore.NewQuery(sql, sql_params)
+	if(!count_query.warn_execute())
+		qdel(count_query)
+		return
+
+	while(count_query.NextRow())
+		var/value = text2num(count_query.item[1])
+		qdel(count_query)
+		return value
+	qdel(count_query)
+
+/*
+  * # getBookRatings
+  *
+  * External proc that gets all of the book ratings for a book. Unless the requested SSID doesn't exist in the
+  * database, this proc will return (if the book has ratings) a list with the
+  * first element being the books avg ratings and a list of ratings by players ["ckey", rating_int]
+  *
+  * Arguments:
+  * * bookid - SSID of the book you wish to get ratings for
+ */
+/datum/library_catalog/proc/getBookRatings(bookid)
+	var/list/sql_params = list()
+	sql_params["id"] = bookid
+
+	var/datum/db_query/query = SSdbcore.NewQuery("SELECT ratings, raters FROM library WHERE id=:id", sql_params)
+	if(!query.warn_execute())
+		qdel(query)
+		return
+
+	var/list/book_ratings = list()
+	while(query.NextRow())
+		if(!query.item[2]) //we don't want to decode something that is null
+			break
+		book_ratings = list(query.item[1], json_decode(query.item[2]))
+
+	qdel(query)
+	return book_ratings
+
+/*
+  * # getRandomBooks
+  *
+  * External proc that gets random books from the Database, used by spawners for the most part. RANT: whoever wrote the fucking
+  * old library code made this a global proc that accepted a loc and new'd/spawned in books from THIS PROC, take this as a
+  * lesson never to do this. Anywho, this proc returns a list of cached books.
+  *
+  * Arguments:
+  * * amount - amount of random books to get
+ */
+/datum/library_catalog/proc/getRandomBooks(amount = 1)
+	if(!amount)
+		return
+	if(!SSdbcore.IsConnected())
+		return
+	var/num_books = clamp(amount, 1, 50) //you don't need more than 50 random books <3
+	var/list/sql_params = list("amount" = num_books )
+	var/sql = "SELECT id, author, title, content, summary, ratings, primaryCategory, secondaryCategory, tertiaryCategory, ckey FROM library GROUP BY title ORDER BY rand() LIMIT :amount"
+	var/datum/db_query/query = SSdbcore.NewQuery(sql, sql_params)
+	if(!query.warn_execute())
```

Comment:
```suggestion
	if(!query.warn_execute(async = FALSE))
```
This needs to not be async, as its called during `Initialize()`, and `Initialize()` should never sleep.